### DISCUSSION TITLE
Help with binding between two controls

### DISCUSSION THREAD

**Discussion by eylvisaker** [AvaloniaUI/Avalonia#10796] [Q&A] [answered]
I'm trying to create a ListBox where the user can select an item from the box and then see the details in a UserControl (ActorCard) next to it. I have a second ListBox there to test the bindings, and the two ListBoxes have their SelectedItem property bound to the same property in my view model. The ListBoxestrack their selected item as expected, if I select an item in one, it becomes selected in the other. But the ActorCard that's in the middle never gets its binding updated.

![image](https://user-images.githubusercontent.com/8533471/227730134-47c47b9c-870a-4189-b351-6eaef0256888.png)

What am I doing wrong here? I've tried applying Mode=TwoWay everywhere but that doesn't seem to help.

Edited to add: Adding definitions of ActorCard I've tried. I've tried with both a DirectProperty and a StyledProperty and neither works, although they behave differently. 

* StyledProperty: the behavior I described above - the ActorCards that are part of the second listbox do get their Actor property assigned when the view is shown, but the binding for the ActorCard in the middle doesn't get set in response to changing the item in the list. By placing a breakpoint in NotifyActorChanged I can see the values get set when the view is first shown, but it's never set in response to clicking a listbox item.
 ```
    public partial class ActorCard : UserControl
    {
        #region --- Actor Property ---

        public static readonly StyledProperty<ActorViewModel> ActorProperty =
            AvaloniaProperty.Register<ActorCard, ActorViewModel>(nameof(Actor),
                notifying: NotifyActorChanged);

        private static void NotifyActorChanged(IAvaloniaObject arg1, bool arg2)
        {
            if (arg1 is ActorCard actorCard)
            {
                actorCard.DataContext = actorCard.Actor;
            }
        }

        public ActorViewModel Actor
        {
            get => GetValue(ActorProperty);
            
```

* DirectProperty: the ActorCards in second listbox don't show anything. I placed a breakpoint in the setter, the value being set is always null, and it never hits the breakpoint when clicking on listbox items.

```
    public partial class ActorCard : UserControl
    {
        #region --- Actor Property ---

        ActorViewModel actor;

        public static readonly DirectProperty<ActorCard, ActorViewModel> ActorProperty =
            AvaloniaProperty.RegisterDirect<ActorCard, ActorViewModel>(nameof(Actor), o => o.Actor, (o, v) => o.Actor = v);

        public ActorViewModel Actor
        {
            get => actor;
            set
            {
                SetAndRaise(ActorProperty, ref actor, value);
                DataContext = actor;
            }
        }
```

**Comment by lukaszossowski** [comment]
How have you checked that it never gets updated? The code here seems fine, I'm guessing the issue might lie in the ActorCard control.

**Reply by eylvisaker** [reply]
I've updated the question with the code from ActorCard. I've tried StyledProperty and DirectProperty and can't get either to work. I can tell it doesn't get updated by setting breakpoints that don't get hit when I click listbox items.

**Comment by timunie** [comment, marked as answer]
Make sure your ViewModel has INPC implemented and that SelectedActor gets notified when changed.

**Reply by lukaszossowski** [reply]
If it weren't implemented the second list wouldn't update though, no?

**Reply by timunie** [reply]
Yep. Exactly 👍

**Reply by eylvisaker** [reply]
The ViewModel for ActorsPage is INPC and I can tell from setting breakpoints that get the notification when it changes, but the binding to the ActorCards doesn't get updated. I've verified that the PropertyChanged event does fire with e.PropertyName == SelectedActor but the binding doesn't update.

**Reply by timunie** [reply]
Why are you even changing the DataContext? That doesn't make sense. I also suggest not to base on UserControl. Here is a sample showing how to create your own re-usable control: https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/CustomControls/RatingControlSample

**Reply by eylvisaker** [reply]
Thanks, that was exactly what I needed!
